cpp/src/arrow/engine/substrait/expression_internal.cc

```diff
@@ -159,21 +159,11 @@ Result<compute::Expression> FromProto(const substrait::Expression& expr,
 
       ARROW_ASSIGN_OR_RAISE(auto decoded_function,
                             ext_set.DecodeFunction(scalar_fn.function_reference()));
+      
+      auto arrow_function = ext_set.GetFunctionMap().GetArrowFromSubstrait(decoded_function.name.to_string());
 
-      std::vector<compute::Expression> arguments(scalar_fn.args_size());
-      for (int i = 0; i < scalar_fn.args_size(); ++i) {
-        ARROW_ASSIGN_OR_RAISE(arguments[i], FromProto(scalar_fn.args(i), ext_set));
-      }
+      return arrow_function(scalar_fn);                      
```

Comment:
What happens if `arrow_function` is not found?

Reply:
Returning a KeyError status if not found.